### PR TITLE
fix(shell): Work around debian dash shell lack of source

### PR DESCRIPTION
**Pull request by bjornsnoen**
Add support for modern versions of ubuntu and debian which use dash as the default shell implementation

### CODE REVIEW

**Comment by sobolevn** [comment]
Does it work for you locally?

File: asdf.py [hunk 1]
```diff
@@ -127,10 +127,11 @@ def _run_command(self, command, message=None, error_message=None, **kwargs):
 
         if self._system_sh_is_dash():
             # dash doesn't overwrite $SHELL so lets try to use that.
-            shell = getenv("SHELL")
+            shell = getenv("SHELL", default="/usr/bin/bash")
             self._log.debug(f"dash detected, attempting to use user shell {shell}")
-            command = f'{shell} -c "{command}"'
+            command = '%s -c "%s"' % (shell, command)
```
Comment (sobolevn) [inline asdf.py:132]: Please, don't use `%s`

Comment (bjornsnoen) [reply asdf.py:132]: Oh, I was using fstrings but I changed it to be more backwards compatible. Should I change back to fstrings or do you prefer something else?

**Comment by bjornsnoen** [comment]
> Does it work for you locally?

Yes, I wrote the code in an ubuntu 23.04 vm which is affected by the issue. Using this code I can use the asdf plugin again.

File: asdf.py (hunk 1 above)
Comment (sobolevn) [reply asdf.py]: `.format` should be the best in this case! 👍

File: asdf.py [hunk 2]
```diff
@@ -104,13 +104,33 @@ def _handle_install(self, data):
...
 
         if self.asdf_location:
-            command = "source %s && %s" % (self.asdf_location, command)
-
+            command = f"source {self.asdf_location} && {command}"
+        if self._system_sh_is_dash():
+            # dash doesn't overwrite $SHELL so lets try to use that.
+            shell = getenv("SHELL", default="/usr/bin/bash")
+            self._log.debug(f"dash detected, attempting to use user shell {shell}")
+            command = f'{shell} -c "{command}"'
```
Comment (sobolevn) [inline asdf.py:131]:
```suggestion
            command = '{0} -c "{1}"'.format(shell, command)
```

**Comment by sobolevn** [comment]
Thanks a lot!

**Pull request merged**